Update rustdoc types

Velfi opened the pull request:
*Description of changes:*

Dependabot got hung up on a `rustdoc-types` breaking change so I did the update myself.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Velfi commented:
I'm going to pick this back up later since I can't remember how to update the rustdoc format version.

jdisanti commented:
You can look at previous upgrade PRs to see all the files that need to get bumped. For example, https://github.com/awslabs/cargo-check-external-types/pull/155/files

Looks like you missed the README and the GitHub Actions CI workflow.

Velfi commented:
> You can look at previous upgrade PRs to see all the files that need to get bumped. For example, https://github.com/awslabs/cargo-check-external-types/pull/155/files
> 
> Looks like you missed the README and the GitHub Actions CI workflow.

Thanks John. I fixed that and added directions for updating to the README.
